pkg/gatewayserver/gatewayserver_test.go

```diff
@@ -874,6 +874,72 @@ func TestGatewayServer(t *testing.T) {
 							}
 						})
 					}
+					t.Run("Disconnection", func(t *testing.T) {
+						for _, tc := range []struct {
+							Name     string
+							Antennas []*ttnpb.GatewayAntenna
+						}{
+							{
+								Name: "NoDisconnect",
+								Antennas: []*ttnpb.GatewayAntenna{
+									{
+										Gain: float32(0),
+									},
+								},
+							},
+							{
+								Name: "Disconnect",
+								Antennas: []*ttnpb.GatewayAntenna{
+									{
+										Gain: float32(3),
+									},
+								},
+							},
+						} {
```

Comment:
You will have to do `ptc := ptc` before `t.Run` until Go 1.22. See [this](https://go.dev/wiki/LoopvarExperiment).